integration-tests/src/test/java/org/hps/test/it/ReconTest.java

```diff
@@ -32,7 +32,7 @@
     /**
      * The default tolerance level for histogram statistics comparison
      */
-    protected static final double DEFAULT_TOLERANCE = 5E-3;
+    protected static final double DEFAULT_TOLERANCE = 1E-6;
```

Comment:
This is unnecessary, lets leave it at at 0.5%.

Reply:
Was 0.5% tolerance arrived at by any rigorous determination?

The idea was that the next PR I put up will allow users to replace the reference plots as needed within their dev branches, so it wouldn't be necessary to have such a generous tolerance on the histogram comparisons, since they wouldn't need to be updated at JLAB any longer.  

Going forward, I'd much rather developers be aware when they change the output at all and then update the files, rather than having a fairly generous tolerance on the test statistics which could mask a series of changes to the recon output. So with this in mind, I'd prefer having the requirement be a nearly exact match between the ref and test output.

Maybe we can discuss further in the software meeting to see if there's a group consensus about it? (The tolerance can be set per-test now as well.)

Reply:
Was 1E-6 arrived at by any rigorous determination? 0.5% is NOT "such a generous tolerance". Half a percent is rather conservative still. There are checks which will still catch if it changes at all in these. The code is currently 0.5% and you are changing it, how about we have a meeting where people to agree to change this?

Reply:
The correct number here is 0...

Reply:
"Was 1E-6 arrived at by any rigorous determination?"

It is a small number that will show you when there is any change in the the test output rather than a fairly large tolerance, at least for certain statistics, that could unintentionally mask changes. I don't remember the current value being discussed really at all, and it was slipped in there within a PR that was ostensibly unrelated without much discussion about it, so I don't think there's really any reason to consider it sacrosanct. The purpose of the tests is showing you when there are any changes to the physics reconstruction output, so the tolerance should be as low as possible. Depending on the plot, 0.5% could be a significant deviation in the mean or RMS, so it is too large of a tolerance for all of the statistics checks to fulfill their overall function.

"how about we have a meeting where people to agree to change this?"

Yes, let's do that...

Reply:
I concur with @bloodyyugo. There is no good way to automate this. Any changes should trigger human intervention with a "physicists" eye to what has changed and why. We should discuss this at the next software meeting.

Reply:
> I concur with @bloodyyugo. There is no good way to automate this. Any changes should trigger human intervention with a "physicists" eye to what has changed and why. We should discuss this at the next software meeting.

We should definitely discuss this in a group setting.

Let me try all the tests with 0 as the tolerance just as a point of reference. 

I believe that will fail currently...

Reply:
Currently the tests fail at less than roughly 1E-6 which was why I committed this number. I can test with a lower tolerance if I update the ref files (but I will wait for a more thorough discussion before doing anything).

Reply:
Since computers are incapable of exactly calculating numbers, and rounding errors can change slightly depending on computer science issues and not physics issues, I disagree whole heartedly. IIRC, this number has never been zero for the relevant assertions. Simple differences in rounding errors can cause this, explicitly the numbers can be different if the references are generated with one system architecture/OS and the tests where the comparison to those references has a different arch/OS. Zero is not the "correct" number because of this and other related reasons. The "CS" eye is more adapted to spotting these issues than a "physicists" eye. Small deviations in the track chi^2 distribution are expected purely by running the tests a different system.

Reply:
A real review process is how a collaborative sw effort should go about understanding and dealing with serious changes.